solana/system_program.py

```diff
@@ -224,12 +224,29 @@ def __check_program_id(program_id: PublicKey) -> None:
 
 def decode_instruction_layout(instruction: TransactionInstruction) -> InstructionLayout:
     """Decode a system instruction and retrieve the instruction layout."""
-    raise NotImplementedError("decode_instruction_layout not implemented")
+    # Slice the first 4 bytes to get the type
+    type_data = instruction.data[:4]
+    type_idx = int.from_bytes(type_data, "little")
+    if 0 <= type_idx and type_idx < len(SYSTEM_INSTRUCTION_LAYOUTS):
+        return SYSTEM_INSTRUCTION_LAYOUTS[type_idx]
+    raise ValueError("Unknow Transaction Instruction")
 
 
 def decode_create_account(instruction: TransactionInstruction) -> CreateAccountParams:
     """Decode a create account system instruction and retrieve the instruction params."""
-    raise NotImplementedError("decode_create_account not implemented")
+    __check_program_id(instruction.program_id)
+    __check_key_length(instruction.keys, 2)
```

Comment:
While I was reviewing this I found `__check_key_length` to be confusing. Could you help me rename this to `__check_num_keys`?

Reply:
sure.